server/welcomebot.go

```diff
@@ -228,12 +230,55 @@ func (p *Plugin) processActionMessage(messageTemplate MessageTemplate, action *A
 }
 
 func (p *Plugin) joinChannel(action *Action, channelName string) {
-	if channel, err := p.API.GetChannelByName(action.Context.TeamID, channelName, false); err == nil {
-		if _, err := p.API.AddChannelMember(channel.Id, action.Context.UserID); err != nil {
-			p.API.LogError("Couldn't add user to the channel, continuing to next channel", "user_id", action.Context.UserID, "channel_id", channel.Id)
+	// If it begins with @ create a DM channel
+	if strings.HasPrefix(channelName, "@") {
+		r := []rune(channelName)
+		dmUser, userErr := p.API.GetUserByUsername(string(r[1:]))
+	
+		if userErr != nil {
+			p.API.LogError("Couldn't find DM user, continuing to next channel", "channelName", channelName)
+			return
+		}
+
+		if !dmUser.IsBot {
+			p.API.LogError("Specified DM user is not a bot, continuing to next channel", "channelName", channelName)
+			return
+		}
+		
+		dmChannel, err := p.API.GetDirectChannel(dmUser.Id, action.Context.UserID)
+
+		if err != nil {
+			p.API.LogError("Couldn't create or get DM channel, continuing to next channel", "user_id", action.Context.UserID, "channelName", channelName, "channel_id", dmChannel.Id)
+			return
+		}
```

Comment:
```suggestion
		dmChannel, err := p.API.GetDirectChannel(dmUser.Id, action.Context.UserID)
		if err != nil {
			p.API.LogError("Couldn't create or get DM channel, continuing to next channel", "user_id", action.Context.UserID, "channelName", channelName, "channel_id", dmChannel.Id)
			return
		}
```

Reply:
What is being suggested here? Removing the newline at line 249?